### PR TITLE
Fix opening image with '%' in the filename

### LINKED ISSUE #83165: An Error occured while loading the image
Issue Type: <b>Bug</b>

I have noticed that every time I try to open an image on vs code, I get the error stated on the title no matter what image file type

VS Code version: Code 1.39.2 (6ab598523be7a800d7f3eb4d92d7ab9a66069390, 2019-10-15T15:33:00.827Z)
OS version: Darwin x64 18.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2700)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (6.19GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (21)</summary>

Extension|Author (truncated)|Version
---|---|---
html-snippets|abu|0.2.1
vscode-browser-preview|auc|0.5.9
rails|bun|0.13.1
vscode-gemfile|bun|0.0.2
solargraph|cas|0.21.1
vscode-eslint|dba|1.9.1
code-runner|for|0.9.14
vscode-rufo|jnb|0.0.1
vscode-gutter-preview|kis|0.23.0
atom-keybindings|ms-|3.0.6
vsliveshare|ms-|1.0.1056
debugger-for-chrome|msj|4.12.0
erb|Rie|0.1.1
LiveServer|rit|5.6.1
ActiveFileInStatusBar|Ros|1.0.3
slim|sia|0.1.2
html-preview-vscode|tht|0.2.5
highlight-matching-tag|vin|0.9.5
simple-ruby-erb|vor|0.2.1
quokka-vscode|Wal|1.0.257
erb-vscode-snippets|Zne|0.0.4

</details>

### PR DESCRIPTION
**Pull request by onequid**
This PR fixes #83165 opening image with '%' in the filename
Please let me know how to write tests for this if it is required as I don't have a clue.
I have no idea in what situation `scheme` would be as 'git' or 'data' neither, please give advice and I would try it out

### CODE REVIEW

**Comment by OneQuid** [comment]
By the way, do you any conventions for `encodeURI` and `decodeURI` thing? 
I am a bit concerned if there would be any other situations that '%' not handled propriately, as I saw functions calling `decodeURIComponent()` themselves and another function calling `decodeURIComponent()`, where the url is decoded twice in a row.

**Comment by mjbvz** [comment]
I believe this was originally added to handle file names with spaces in the name. Can you please confirm images with names like `a b.png` still load after this change?

File: extensions/image-preview/src/preview.ts [hunk 1]
```diff
@@ -223,15 +223,15 @@ class Preview extends Disposable {
 	private getResourcePath(webviewEditor: vscode.WebviewPanel, resource: vscode.Uri, version: string) {
 		switch (resource.scheme) {
 			case 'data':
```
Comment (OneQuid) [inline extensions/image-preview/src/preview.ts]: What user operation or situation would run codes in the `case 'data'`?
I guess that the `case 'git'` part would be fine. Not sure about the `case 'data'` section.

**Comment by OneQuid** [comment]
I see, I didn't think of the spaces.
And I just tried and glad to say that it worked good.
screenshot

I am still concerned about another `case` section.
Could you please share some more of your knowledge?
I add a review comment just in case I did not make it clear.

**Comment by mjbvz** [comment]
Thanks! This will be in the next insiders build and is scheduled to be shipped with VS Code 1.41

**Comment by OneQuid** [comment]
Glad to know that! My pleasure! Thanks!

**Comment by mjbvz** [comment]
This broke loading of images with spaces in the name. Given that these are more common than images with `%`, I'm reverting this change

**Comment by OneQuid** [comment]
I assume that both PR (this and #84334 ) would locate the same cause, and only one of them should be merged. It is more about which to merge.

Merging this PR solely won't break anything as far as I tested, in my defence. 
Regards

**Pull request merged**